src/server.c

```diff
@@ -1609,22 +1609,6 @@ void whileBlockedCron(void) {
     mstime_t latency;
     latencyStartMonitor(latency);
 
-    /* In some cases we may be called with big intervals, so we may need to do
-     * extra work here. This is because some of the functions in serverCron rely
-     * on the fact that it is performed every 10 ms or so. For instance, if
-     * activeDefragCycle needs to utilize 25% cpu, it will utilize 2.5ms, so we
-     * need to call it multiple times. */
-    long hz_ms = 1000 / server.hz;
-    while (server.blocked_last_cron < server.mstime) {
-        /* Defrag keys gradually. */
-        activeDefragCycle();
```

Comment:
We should still have a hook here to ensure that we don't have defrag starvation.  Timers are not fired while blocked.  In a degenerative case - let's say an hour long script - we don't want to become completely unresponsive for 6 minutes after the script ends to achieve 10% CPU.